code/parachain/frame/composable-maths/src/dex/constant_product.rs

```diff
@@ -465,23 +465,29 @@ pub fn compute_deposit_lp(
 ///
 /// https://github.com/ComposableFi/composable/blob/main/rfcs/0008-pablo-lbp-cpp-restructure.md#42-liquidity-provider-token-lpt-math-updates
 /// Equation 6
-pub fn compute_first_deposit_lp_<T: PerThing>(
-	pool_assets: &[(u128, u128, T)],
+pub fn compute_first_deposit_lp_<T, I>(
+	pool_assets: I,
 	_f: T,
-) -> ConstantProductAmmResult<ConstantProductAmmValueFeePair> {
+) -> ConstantProductAmmResult<ConstantProductAmmValueFeePair>
+where
+	T: PerThing,
+	I: IntoIterator<Item = ConstantProductFirstDepositInput<T>>,
+	I::IntoIter: ExactSizeIterator,
```

Comment:
I also refactored this fn to be a bit more general, IMO this is a much cleaner interface.